Code/src/main/java/nl/utwente/group10/ui/components/OutputAnchor.java

```diff
@@ -2,13 +2,52 @@
 
 import javafx.scene.input.MouseEvent;
 import nl.utwente.group10.ui.CustomUIPane;
+import nl.utwente.group10.ui.handlers.OutputAnchorHandler;
 
 import java.io.IOException;
+import java.util.Optional;
 
-public class OutputAnchor extends ConnectionAnchor {
+public class OutputAnchor extends ConnectionAnchor{
     public OutputAnchor(Block block, CustomUIPane pane) throws IOException {
         super(block, pane);
+        new OutputAnchorHandler(pane.getConnectionCreationManager(),this);
+    }
+
+	@Override
+	public Optional<Connection> createConnectionWith(ConnectionAnchor other) {
+		if(other instanceof InputAnchor){
+	    	return Optional.of(createConnectionTo((InputAnchor) other));
+		}else{
+			return Optional.empty();
+		}
+	}
+	
+	public Connection createConnectionTo(InputAnchor other){
+		new Connection(this, other);
+		getPane().getChildren().add(getConnection().get());
+		getPane().invalidate();
+		return getConnection().get();
+	}
+	
+	@Override
+	public String toString(){
+    	return "OutputAnchor belonging to "+getBlock().getName();
+    }
+
+	@Override
+	public boolean canConnect() {
+		//OutputAnchors can have multiple connections;
+		return true;
+	}
 
-        this.addEventHandler(MouseEvent.MOUSE_CLICKED, event -> pane.setLastOutputAnchor(this));
+	@Override
+	public Optional<Connection> getConnection(){
+    	//Does not keep track of its connections.
```

Comment:
In theorie *hoeft* een OutputAnchor niks te weten van welke verbindingen er aan hangen: blokken vragen aan hun inputanchors wat hun waarde moet zijn, niet vice-versa. Vandaar dat ik OutputAnchor geen connection (of verzameling connections) liet bijhouden. Door getConnection naar de abstracte superklasse te verplaatsen zeg je dat elk anker al zijn verbindingen bij moet gaan houden; is dat inderdaad het plan?